### PR TITLE
Improvments for less drain of battery and during reconnection

### PR DESCRIPTION
**Pull request by miloit**
Changed the mechanism of reconnection
Delete the polling timer to drain less batterie

### CODE REVIEW

File: src/openhab.cpp [hunk 1]
```diff
@@ -224,63 +211,63 @@ void OpenHAB::connect() {
...
+            QObject::connect(&_sseNetworkManager, &QNetworkAccessManager::finished, this, &OpenHAB::streamFinished);
+            QObject::connect(_sseReconnectTimer, &QTimer::timeout, this, &OpenHAB::onSseTimeout);
+
+            startSse();
+            getItems(true);
+            setState(CONNECTED);
         }
+        QObject::disconnect(&_nam, &QNetworkAccessManager::finished, this, 0);
     });
-    _checkProcessOpenHabAvailability.start("curl", QStringList() << "-s" << _url);
```
Comment (zehnm) [inline src/openhab.cpp:229]: Thank you for removing this ugly hack! :+1:

File: src/openhab.cpp [hunk 2]
```diff
@@ -224,63 +211,63 @@ void OpenHAB::connect() {
             _flagOpenHabConnected = false;
             disconnect();
             qCDebug(m_logCategory) << "openhab not reachable";
+        } else if (reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt() == 200) {
+            qCDebug(m_logCategory) << reply;
+            qCDebug(m_logCategory) << reply->header(QNetworkRequest::ContentTypeHeader).toString();
+            qCDebug(m_logCategory) << reply->header(QNetworkRequest::LastModifiedHeader).toDateTime().toString();;
+            qCDebug(m_logCategory) << reply->header(QNetworkRequest::ContentLengthHeader).toULongLong();
+            qCDebug(m_logCategory) << reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+            qCDebug(m_logCategory) << reply->attribute(QNetworkRequest::HttpReasonPhraseAttribute).toString();
```
Comment (zehnm) [inline src/openhab.cpp]: Please merge into a single qCDebug statement (<< can be chained).

File: src/openhab.cpp [hunk 3]
```diff
@@ -192,28 +191,16 @@ void OpenHAB::connect() {
...
-            setState(CONNECTED);
-            QObject::disconnect(&_checkProcessOpenHabAvailability,
-                                static_cast<void(QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished), 0, 0);
-        } else {
+    QNetworkRequest request(_url + "/systeminfo");
+    request.setHeader(QNetworkRequest::ContentTypeHeader, "text/plain");
+    if (_token != "") {
+        request.setRawHeader("accept", "*/*");
+        QString token = "Bearer " +_token;
+        request.setRawHeader(QByteArray("Authorization"), token.toUtf8());
```
Comment (zehnm) [inline src/openhab.cpp]: `QByteArray` can be omitted, it's automatically created.

File: src/openhab.cpp [hunk 4]
```diff
@@ -413,8 +375,13 @@ void OpenHAB::processItems(const QJsonDocument& result, bool first) {
             }
         }
         if ((_myEntities.count() - countFound) > 0) {
-            m_notifications->add(true,
-                                 "openHAB - entities missing : " + QString::number((_myEntities.count() - countFound)));
+            m_notifications->add(
+                        true, "Could not load : " + QString::number((_myEntities.count() - countFound))
+                        + "openHAB items", tr("Reconnect"), [](QObject* param) {
+                Integration* i = qobject_cast<Integration*>(param);
+                i->connect();
```
Comment (zehnm) [inline src/openhab.cpp]: Does it make sense to offer the user to reconnect to OH?
If I understand this correctly, then this is only a notification message that items are missing, but the connection itself is still valid.

File: src/openhab.cpp [hunk 5]
```diff
@@ -616,3 +576,30 @@ void OpenHAB::sendOpenHABCommand(const QString& itemId, const QString& state) {
...
+        if (reply->error()) {
+            qCCritical(m_logCategory) << reply->errorString();
+        } else {
+            qCDebug(m_logCategory) << reply;
+            QString answer = reply->readAll();
+            qCDebug(m_logCategory) << reply->header(QNetworkRequest::ContentTypeHeader).toString();
+            qCDebug(m_logCategory) << reply->header(QNetworkRequest::LastModifiedHeader).toDateTime().toString();;
+            qCDebug(m_logCategory) << reply->header(QNetworkRequest::ContentLengthHeader).toULongLong();
+            qCDebug(m_logCategory) << reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+            qCDebug(m_logCategory) << reply->attribute(QNetworkRequest::HttpReasonPhraseAttribute).toString();
```
Comment (zehnm) [inline src/openhab.cpp]: See above: single qCDebug statement

**Review by zehnm — CHANGES_REQUESTED** [review]
Thank you for the improvements, especially for getting rid of the curl execution.
Just a few minor things, otherwise I can't say much to the OH functionality. Haven't used it so far.

File: src/openhab.cpp (hunk 1 above)
Comment (miloit) [reply src/openhab.cpp:229]: :)

File: src/openhab.cpp (hunk 3 above)
Comment (miloit) [reply src/openhab.cpp]: Done

File: src/openhab.cpp (hunk 2 above)
Comment (miloit) [reply src/openhab.cpp]: removed the unnecessary information

File: src/openhab.cpp (hunk 5 above)
Comment (miloit) [reply src/openhab.cpp]: removed as well

File: src/openhab.cpp (hunk 4 above)
Comment (miloit) [reply src/openhab.cpp]: That's true. My idea was to give the user the change to reconnect but it#s just an information. Changed it back.

**Review by zehnm — APPROVED** [review]

**Pull request merged**